File: Reservations/Reservations.Domain/EventOccurrence.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Reservations.Domain
{
    /// <summary>
    /// Event instance and it's date
    /// </summary>
    public class EventOccurrence : AuditableEntity
    { 
        public EventOccurrence()
        {
            Tickets = new HashSet<Ticket>();
        }
        public long Id { get; set; }
        public DateTime StartTime { get; set; }
        public long EventId { get; set; }
        public Event Event { get; set; }
        public ICollection<Ticket> Tickets { get; set; }

    }
}

File: reservation-frontend/src/app/models/eventOccurrence.ts
import { MatTableDataSource } from "@angular/material/table"
import { Event } from "./event"
import { Reservation } from "./reservation"
import { Ticket } from "./ticket"

export class EventOccurrence {
    public id!: number
    public startTime!: Date
    public eventId!: number
    public event!: Event
    public tickets!: Ticket[]
    public totalTicketCount!: number
    public availableTicketCount!: number
    public reservedTicketCount!: number
    public soldTicketCount!: number
    public reservations!: Reservation[]
    public isActive!: boolean
    public isDeleted!: boolean
}